docs/advanced/root-ca-certificates.md

```diff
@@ -0,0 +1,72 @@
+
+New root certificates can be [installed into a custom image](https://stackoverflow.com/questions/42292444/how-do-i-add-a-ca-root-certificate-inside-a-docker-image) using standard OS packages. Then, this custom image can be used [to override the base image for `ko`](https://ko.build/configuration/#overriding-base-images). Once the Go application container image is built using `ko` with the custom base image, the root certificates installed on the base image will be trusted by the Go application.
+
+### Example
+
+1. Make a custom container image with your new root certificates
+```dockerfile
+# Dockerfile
+FROM alpine
+
+RUN apk update
+RUN apk add ca-certificates
+
+ADD new-root-ca.crt /usr/local/share/ca-certificates/new-root-ca.crt
+RUN chmod 644 /usr/local/share/ca-certificates/new-root-ca.crt
+RUN update-ca-certificates
+```
+
+2. Build and push the custom container image to a container registry
+```sh
+docker build . -t docker.io/ko-build/image-with-new-root-certs
+docker push docker.io/ko-build/image-with-new-root-certs
+```
+
+3. Configure `ko` to [override the default base image](https://ko.build/configuration/#overriding-base-images) with the custom image
+```yaml
+# .ko.yaml
+defaultBaseImage: docker.io/ko-build/image-with-new-root-certs
+```
+
+    **OR**
+```sh
+export KO_DEFAULTBASEIMAGE=docker.io/ko-build/image-with-new-root-certs
+```
+
+4. Build the Go app container image with `ko`
+```sh
+ko build .
+```
+
+## Static Assets
+Alternatively, root certificates can be installed into the Go application container image using a combination of [`ko` static assets](https://ko.build/features/static-assets/) and [overriding the default system location for SSL certificates](https://pkg.go.dev/crypto/x509#SystemCertPool).
+
+Using `ko`'s support for static assets, root certificates can be stored in the `<importpath>/kodata` directory (either checked into the repository, or injected dynamically by a CI pipeline). After running `ko build`, the certificate files are then bundled into the built image at the path `$KO_DATA_PATH`.
+
+To enable the Go application to trust the bundled certificate(s), the container runtime or orchestrator (Docker, Kubernetes, etc) must set the environment variable `SSL_CERT_DIR` to the same value as `KO_DATA_PATH`. Go [uses `SSL_CERT_DIR` to determine the directory to check for SSL certificate files](https://go.dev/src/crypto/x509/root_unix.go). Once this variable is set, the Go application will trust the bundled root certificates in `$KO_DATA_PATH`.
+
+### Example
+
+1. Copy the root certificate(s) to the `<importpath>/kodata/` directory
+```sh
+# $(pwd) assumed to be at <importpath> for this example
+mkdir -p kodata
+cp $CERT_FILE_DIR/*.crt kodata/
+```
+
+2. Build the Go application container image
+```sh
+KO_DOCKER_REPO=docker.io/ko-build/static-assets-certs ko build .
+```
+
+3. Run the Go application container image with `SSL_CERT_DIR` equal to `/var/run/ko` (the default value for `$KO_DATA_PATH`)
```

Comment:
**question**: I thought of something I might have misunderstood while writing this PR.

Can `ko` users override the value for `KO_DATA_PATH` and subsequently where `ko` will bundle the static assets in the built image? I tried doing this myself and could not override the default path of `/var/run/ko` on a Linux base image.

After looking at [the code](https://github.com/ko-build/ko/blob/903da574dc54a272a58b3881dee1f61a4a284b1f/pkg/build/gobuild.go#L775) I think I understand that this value is statically set, and is abstracted via an environment variable just for the sake of running `ko` on different operating systems.

Is this correct? If so, I will probably change this example to say 
> equal to `/var/run/ko` (the default value for `$KO_DATA_PATH` on Linux)

Also, if I am correct about the value being statically set per OS, should we clarify that on the [static assets documentation page](https://ko.build/features/static-assets/)?

Reply:
When running `ko` to build, the directory is by convention `<importpath>/kodata` with no way to override that.

The env var comes into play _inside_ the container when it runs, and points to the path where the contents of `kodata` have been bundled in the image (`/var/run/ko`). The reason the env var is useful is so that you can have code that works inside and outside the container:

```
    http.Handle("/", http.FileServer(http.Dir(os.Getenv("KO_DATA_PATH"))))
```

works in the running container since `ko` sets that env for you to `/var/run/ko`, and outside the container you can do `KO_DATA_PATH=cmd/foo/kodata go run ./cmd/foo` to achieve the same result.

Let me know if there's a clearer way to state that in the docs.